wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -730,34 +729,40 @@ private void printDetailedTable(List<List<String>> detailList, int columnSize, S
         getLog().info("");
     }
 
-    private void runLocalServer(String projectRootPath) {
+    private void setupLocalServer(String projectRootPath) {
         try {
-            String userHome;
-            if (System.getProperty(Constants.OS_TYPE).toLowerCase().contains(Constants.OS_WINDOWS)) {
-                userHome = System.getenv(Constants.USER_PROFILE);
-            } else {
-                userHome = System.getenv(Constants.HOME);
-            }
-            Path miDownloadPath = Paths.get(userHome, Constants.M2, Constants.REPOSITORY, Constants.ORG,
-                    Constants.WSO2, Constants.EI, Constants.WSO2_MI, server.getServerVersion());
-            URL downloadUrl;
+            URL downloadUrl = new URL(baseUrl + server.getServerVersion() + Constants.SLASH_WSO2_MI_WITH_DASH +
+                    server.getServerVersion() + Constants.ZIP);
+            String userHome = getUserHome();
+            Path miDownloadPath = Paths.get(userHome, Constants.WSO2_MI, Constants.MICRO_INTEGRATOR);
+            Path fullFilePath = Paths.get(miDownloadPath.toString(), Constants.WSO2_MI_WITH_DASH +
+                    server.getServerVersion());
+            Path zipFilePath = Paths.get(fullFilePath + Constants.ZIP);
             if (server.getServerDownloadLink() == null) {
-                downloadUrl = new URL("https://github.com/wso2/micro-integrator/releases/download/v" +
-                        server.getServerVersion() + "/wso2mi-" + server.getServerVersion() + Constants.ZIP);
+                if (server.getServerVersion().equals(Constants.MI_4_4_0)) {
+                    downloadUrl = new URL(baseUrl + server.getServerVersion() + Constants.SLASH_WSO2_MI_WITH_DASH
+                            + server.getServerVersion() + Constants.UPDATED + Constants.ZIP);
```

Comment:
For 4.3.0 will the downloadUrl work as it does not have a "updated" substring in its url?